### PR TITLE
xdrgen: remove gemfile and rakefile to just use docker for the xdrgen

### PR DESCRIPTION
**Pull request by paulbellamy**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Remove `Gemfile*`, and `Rakefile`, and replace with a docker image in `Makefile`

### Why

Inspired by: https://github.com/stellar/rs-stellar-xdr/blob/main/Makefile#L11-L26

Reduces some code duplication, and "ruby-ness"

### Known limitations

Doesn't have a rake command to copy over xdr files from stellar-core anymore. Could add a make rule for that if we feel strongly about it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,24 @@ gxdr/xdr_generated.go: $(XDRS)
 	go run github.com/xdrpp/goxdr/cmd/goxdr -p gxdr -enum-comments -o $@ $(XDRS)
 	go fmt $@
 
-xdr/xdr_generated.go: $(XDRS) Rakefile Gemfile.lock
-	bundle exec rake xdr:generate
+define xdrheader
+//lint:file-ignore S1005 The issue should be fixed in xdrgen. Unfortunately, there's no way to ignore a single file in staticcheck.
+//lint:file-ignore U1000 fmtTest is not needed anywhere, should be removed in xdrgen.
```
Comment (leighmcculloch) [inline Makefile:37]: 💡 I think we should add these to the frontmatter of the go generator in xdrgen. It'll make this simpler, and it'll make it easier to use xdrgen with other xdr. If you do this, it will simplify this Makefile target I think.

File: Makefile [hunk 2]
```diff
@@ -32,7 +32,24 @@ gxdr/xdr_generated.go: $(XDRS)
...
+endef
+export xdrheader
+
+xdr/xdr_generated.go: $(XDRS)
+	docker run --platform linux/amd64 -it --rm -v $$PWD:/wd -w /wd ruby /bin/bash -c '\
```
Comment (leighmcculloch) [inline Makefile:42]: You shouldn't need the platform specification here. I use the arm64 ruby image to do this and it works fine.
```suggestion
	docker run -it --rm -v $$PWD:/wd -w /wd ruby /bin/bash -c '\
```

File: Makefile [hunk 3]
```diff
@@ -32,7 +32,24 @@ gxdr/xdr_generated.go: $(XDRS)
...
+		gem install specific_install && \
```
Comment (leighmcculloch) [inline Makefile:43]: Could you lock this gem install to a specific version to ensure consistency.
```suggestion
		gem install specific_install -v 0.3.7 && \
```

Fun fact, this still doesn't guarantee a consistent install of dependencies, and there's no way with `gem` to make it consistent. (See https://twitter.com/___leigh___/status/1520997579772678144.) It doesn't matter for specific_install though because it has no dependencies.

File: Makefile (hunk 1 above)
Comment (leighmcculloch) [reply Makefile:37]: If you open a PR on xdrgen for this, assign me and I'll approve it.

File: Makefile [hunk 4]
```diff
@@ -32,7 +32,24 @@ gxdr/xdr_generated.go: $(XDRS)
...
 
-xdr/xdr_generated.go: $(XDRS) Rakefile Gemfile.lock
-	bundle exec rake xdr:generate
+define xdrheader
+//lint:file-ignore S1005 The issue should be fixed in xdrgen. Unfortunately, there's no way to ignore a single file in staticcheck.
+//lint:file-ignore U1000 fmtTest is not needed anywhere, should be removed in xdrgen.
+endef
+export xdrheader
+
+xdr/xdr_generated.go: $(XDRS)
```
Comment (leighmcculloch) [inline Makefile:41]: We need a target that downloads the XDRs. The Rakefile does that. This could be some simple curl commands.

File: Makefile (hunk 2 above)
Comment (paulbellamy) [reply Makefile:42]: Ah, hmm. I got a weird error locally.. let me try again.

File: Makefile (hunk 4 above)
Comment (leighmcculloch) [reply Makefile:41]: You could copy this: https://github.com/stellar/rs-stellar-xdr/pull/34

File: Makefile (hunk 1 above)
Comment (paulbellamy) [reply Makefile:40]: https://github.com/stellar/xdrgen/pull/82

**Review by leighmcculloch — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -32,7 +32,23 @@ gxdr/xdr_generated.go: $(XDRS)
...
 
-xdr/xdr_generated.go: $(XDRS) Rakefile Gemfile.lock
-	bundle exec rake xdr:generate
+xdr/%.x:
+	curl -Lsf -o $@ https://raw.githubusercontent.com/stellar/stellar-core/master/src/$@
+
+xdr/xdr_generated.go: $(XDRS)
+	docker run -it --rm -v $$PWD:/wd -w /wd ruby /bin/bash -c '\
+		gem install specific_install -v 0.3.7 && \
+		gem specific_install https://github.com/stellar/xdrgen.git -b master && \
```
Comment (leighmcculloch) [inline Makefile:41]: One more thing, we should probably lock this to a commit of xdrgen.

File: Makefile [hunk 6]
```diff
@@ -32,7 +32,23 @@ gxdr/xdr_generated.go: $(XDRS)
 	go run github.com/xdrpp/goxdr/cmd/goxdr -p gxdr -enum-comments -o $@ $(XDRS)
 	go fmt $@
 
-xdr/xdr_generated.go: $(XDRS) Rakefile Gemfile.lock
-	bundle exec rake xdr:generate
+xdr/%.x:
```
[the remark this answers is not in the document]
Comment (paulbellamy) [reply Makefile:35]: It is like a "resolver". So when `xdr/xdr_generated.go` requires them as dependencies, the files will be downloaded.

[the remark this answers is not in the document]
Comment (paulbellamy) [reply Makefile:35]: this is what the `xdr-update` rule uses to update them. Delete them all, and build `xdr/xdr_generated.go`. That implicitly downloads them.
